src/wallet/wallet2_basic/wallet2_storage.cpp

```diff
@@ -0,0 +1,796 @@
+        cryptonote::account_base decrypted_account;
+        CHECK_AND_ASSERT_THROW_MES(
+            epee::serialization::load_t_from_binary(decrypted_account, account_data),
+            "Could not parse account keys from EPEE binary");
+        if (account_keys_are_encrypted)
+            decrypted_account.decrypt_keys(keys_key);
+        assign_when_mutable(kd.m_account, decrypted_account);
+    }
+
+    ADAPT_JSON_FIELD(nettype, Uint, kd.m_nettype == cryptonote::UNDEFINED);
+    CHECK_AND_ASSERT_THROW_MES(
+        kd.m_nettype == cryptonote::MAINNET ||
+        kd.m_nettype == cryptonote::TESTNET ||
+        kd.m_nettype == cryptonote::STAGENET ||
+        kd.m_nettype == cryptonote::FAKECHAIN,
+        "unrecognized network type for keys_data");
+
+    ADAPT_JSON_FIELD(multisig, Int, false);
+    ADAPT_JSON_FIELD(multisig_threshold, Uint, kd.m_multisig);
+    ADAPT_JSON_FIELD(multisig_rounds_passed, Uint, false);
+    ADAPT_JSON_FIELD(enable_multisig, Int, false);
+    ADAPT_JSON_FIELD(multisig_signers, binary_archive, kd.m_multisig);
+    ADAPT_JSON_FIELD(multisig_derivations, binary_archive, false);
+
+    ADAPT_JSON_FIELD(watch_only, Int, false);
+    ADAPT_JSON_FIELD(confirm_non_default_ring_size, Int, false);
+    ADAPT_JSON_FIELD(ask_password, Int, false); // @TODO: Check AskPasswordType
+    ADAPT_JSON_FIELD(refresh_type, Int, false); // @TODO: Check RefreshType
+    ADAPT_JSON_FIELD(skip_to_height, Uint64, false);
+    ADAPT_JSON_FIELD(max_reorg_depth, Uint64, false);
+    ADAPT_JSON_FIELD(min_output_count, Uint, false);
+    ADAPT_JSON_FIELD(min_output_value, Uint64, false);
+    ADAPT_JSON_FIELD(merge_destinations, Int, false);
+    ADAPT_JSON_FIELD(confirm_backlog, Int, false);
+    ADAPT_JSON_FIELD(confirm_backlog_threshold, Uint, false);
+    ADAPT_JSON_FIELD(confirm_export_overwrite, Int, false);
+    ADAPT_JSON_FIELD(auto_low_priority, Int, false);
+    ADAPT_JSON_FIELD(confirm_export_overwrite, Int, false);
+    ADAPT_JSON_FIELD(segregate_pre_fork_outputs, Int, false);
+    ADAPT_JSON_FIELD(key_reuse_mitigation2, Int, false);
+    ADAPT_JSON_FIELD(segregation_height, Uint, false);
+    ADAPT_JSON_FIELD(ignore_fractional_outputs, Int, false);
+    ADAPT_JSON_FIELD(ignore_outputs_above, Uint64, false);
+    ADAPT_JSON_FIELD(ignore_outputs_below, Uint64, false);
+    ADAPT_JSON_FIELD(track_uses, Int, false);
+    ADAPT_JSON_FIELD(show_wallet_name_when_locked, Int, false);
+    ADAPT_JSON_FIELD(inactivity_lock_timeout, Uint, false);
+    ADAPT_JSON_FIELD(setup_background_mining, Int, false);
+    ADAPT_JSON_FIELD(subaddress_lookahead_major, Uint, false);
+    ADAPT_JSON_FIELD(subaddress_lookahead_minor, Uint, false);
+    ADAPT_JSON_FIELD(always_confirm_transfers, Int, false);
+    ADAPT_JSON_FIELD(print_ring_members, Int, false);
+    ADAPT_JSON_FIELD(store_tx_info, Int, false);
+    ADAPT_JSON_FIELD(default_mixin, Uint, false);
+    ADAPT_JSON_FIELD(export_format, Int, false); // @TODO Check ExportFormat
```

Comment:
I think now I found a small thing that is not implemented. As reported [here](https://github.com/seraphis-migration/monero/pull/4#issuecomment-1731384481) it looked as if loading a .keys file in ASCII format is not yet supported. Now, looking at your code, I think I confirmed that, because support for `ExportFormat::Ascii` is nowhere to be found.

Reply:
Okay, the wallet now supports loading the keys using PEM ascii format, automatically detecting it.

Reply:
(and storing it)